storm-client/src/jvm/org/apache/storm/utils/DisruptorQueue.java

```diff
@@ -502,15 +502,26 @@ private static Long getId() {
 
     private void publishDirectSingle(Object obj, boolean block) throws InsufficientCapacityException {
         long at;
+        long numberOfTuples;
         if (block) {
             at = _buffer.next();
         } else {
             at = _buffer.tryNext();
         }
         AtomicReference<Object> m = _buffer.get(at);
         m.set(obj);
+        if (obj instanceof ArrayList) {
```

Comment:
Seems like the logic for counting number of tuples is duplicated so better to extract method.